packages/bus-dereference/lib/ActorDereferenceBase.ts

```diff
@@ -0,0 +1,58 @@
+import { Readable } from 'stream';
+import { KeysInitQuery } from '@comunica/context-entries';
+import type { IAction, IActorArgs, IActorTest } from '@comunica/core';
+import { Actor } from '@comunica/core';
+import type { IActionContext } from '@comunica/types';
+import type { IActorDereferenceOutput } from '.';
+
+export function emptyReadable<S extends Readable>(): S {
+  const data = new Readable();
+  data.push(null);
+  return <S> data;
+}
+
+/**
+ * Check if hard errors should occur on HTTP or parse errors.
+ * @param {IActionDereference} action A dereference action.
+ * @return {boolean} If hard errors are enabled.
+ */
+export function isHardError(context: IActionContext): boolean {
+  return !context.get(KeysInitQuery.lenient);
+}
+
+/**
+ * A base actor for dereferencing URLs to (generic) streams.
+ *
+ * Actor types:
+ * * Input:  IActionDereference:      A URL.
+ * * Test:   <none>
+ * * Output: IActorDereferenceOutput: A Readable stream
+ *
+ * @see IActionDereference
+ * @see IActorDereferenceOutput
+ */
+export abstract class ActorDereferenceBase<I extends IAction, T extends IActorTest, O extends IActorDereferenceOutput>
```

Comment:
Is there a reason why this one is not just `ActorDereference`?

Reply:
`ActorDereference` and `ActorDereferenceParse` both extend off this base class which is just to provide the `defererenceErrorHandler` to both of them.

I would not expect any actors to extend of `ActorDereferenceBase` - rather they would be extending off the current `ActorDereference`

Reply:
Would it make sense to move `defererenceErrorHandler` into `ActorDereference`, and let `ActorDereferenceParse` from `ActorDereference` directly?

Reply:
ActorDereferenceParse doesn't extend from ActorDereference because they have different ActorOutput's.

ActorDereferenceParse is what used to be ActorAbstraftDereferenceParse before that package was deleted in favour of moving its functionality to this bus